### PR TITLE
Total reqs sent counted

### LINKED ISSUE #71: count the number of requests made
And expose it :).

cc @thekemkid

### PR DESCRIPTION
**Pull request by thekemkid**
Fixes #71

### CODE REVIEW

File: lib/run.js [hunk 1]
```diff
@@ -155,9 +160,10 @@ function run (opts, cb) {
       let result = {
         title: opts.title,
         url: opts.url,
-        requests: histAsObj(requests, totalRequests),
+        requests: histAsObj(requests, totalCompletedRequests),
         latency: addPercentiles(latencies, histAsObj(latencies)),
         throughput: histAsObj(throughput, totalBytes),
+        totalRequests: totalRequests,
```
Comment (mcollina) [inline lib/run.js]: I would put it inside the requests object, as `sent`. Does it make sense?

@davidmarkclements ?

File: lib/run.js [hunk 2]
```diff
@@ -91,8 +92,12 @@ function run (opts, cb) {
     client.on('response', onResponse)
     client.on('connError', onError)
     client.on('timeout', onTimeout)
+    client.on('request', () => { totalRequests++ })
     client.on('done', onDone)
     clients.push(client)
+
+    // we will miss the initial request emits because the client emits request on construction
+    totalRequests += url.pipelining < url.rate ? url.rate : url.pipelining
```
Comment (mcollina) [inline lib/run.js]: nice one!

**Comment by mcollina** [comment]
Does it affect performance?

**Comment by thekemkid** [comment]
No perf damage

File: lib/run.js (hunk 1 above)
Comment (thekemkid) [inline lib/run.js]: I'll rename

File: lib/run.js (hunk 2 above)
Comment (thekemkid) [inline lib/run.js]: I could also trigger the initial request batch by putting it in a nextTick, if you'd prefer

Comment (mcollina) [inline lib/run.js]: This is fine :)

**Comment by mcollina** [comment]
builds are failing

**Comment by thekemkid** [comment]
I went to grab food after updating the var, just realised I never updated
tests 😬 my bad, will fix asap

File: lib/run.js [hunk 3]
```diff
@@ -155,9 +160,10 @@ function run (opts, cb) {
       let result = {
         title: opts.title,
         url: opts.url,
-        requests: histAsObj(requests, totalRequests),
+        requests: histAsObj(requests, totalCompletedRequests),
         latency: addPercentiles(latencies, histAsObj(latencies)),
         throughput: histAsObj(throughput, totalBytes),
+        sent: totalRequests,
```
Comment (mcollina) [inline lib/run.js]: what I meant was `result.requests.sent` :).

Comment (thekemkid) [inline lib/run.js]: Ohh. I forgot to put the helper lib hdr-histogram-percentiles-obj in, is it okay if I do that too?

Comment (mcollina) [inline lib/run.js]: please do!

**Comment by mcollina** [comment]
LGTM, I'd release as minor

**Pull request merged**